tests/test_s3.py

```diff
@@ -970,21 +970,11 @@ def setUp(self):
         ignore_resource_warnings()
         _resource('s3').create_bucket(Bucket=BUCKET_NAME).wait_until_exists()
 
-    @pytest.mark.skipif(condition=sys.platform == 'win32', reason="does not run on windows")
-    @pytest.mark.xfail(
-        condition=sys.platform == 'darwin',
-        reason="MacOS uses spawn rather than fork for multiprocessing",
-    )
```

Comment:
3.14 changes default start methods: https://docs.python.org/3.14/library/multiprocessing.html#multiprocessing-start-methods

since iter_bucket is network bound and not CPU bound, this PR refactors to always use a single shared `concurrent.futures.ThreadPoolExecutor` with a single shared thread-safe boto `S3.Client`.

This makes the code simpler, more predictable, easier to maintain, and cross-platform compatible.

The default of 16 * 4 = global maximum 64 threads is tested to be stable on an EC2 c-family machine downloading from an S3 bucket in the same region.